site/content/en/latest/user/private-key-provider.md

```diff
@@ -0,0 +1,306 @@
+
+## Prerequisites
+
+### For QAT
+
+- Install Linux kernel 5.17 or similar
+- Ensure the node has QAT devices by checking the QAT physical function devices presented.
+
+  ```shell
+  lspci -d 8086:4940
+  ```
+
+- Enable IOMMU from BIOS
+- Enable IOMMU for Linux kernel
+
+  ```shell
+  cat /etc/default/grub:
+  GRUB_CMDLINE_LINUX="intel_iommu=on vfio-pci.ids=8086:4941"
+  update-grub
+  reboot
+  ````
+
+  Once the system is rebooted, check if the IOMMU has been enabled via the following command:
+
+  ```shell
+  dmesg| grep IOMMU
+  [    1.528237] DMAR: IOMMU enabled
+  ```
+
+- Enable virtual function devices for QAT device
+
+  ```shell
+  rmmod qat_4xxx
+  modprobe qat_4xxx
+  qat_device=$(lspci -D -d :4940 | awk '{print $1}')
+  for i in $qat_device; do echo 16|sudo tee /sys/bus/pci/devices/$i/sriov_numvfs; done
+  ```
+
+- Enhance the container runtime memory lock limit
+
+  ```shell
+  mkdir /etc/systemd/system/containerd.service.d
+  cat <<EOF >>/etc/systemd/system/containerd.service.d/memlock.conf
+  [Service]
+  LimitMEMLOCK=134217728
+  EOF
+  ```
+
+  Restart the container runtime (for containerd, CRIO has similar concept)
+
+  ```shell
+  systemctl daemon-reload
+  systemctl restart containerd
+  ```
+
+- Install [Intel® QAT Device Plugin for Kubernetes](https://github.com/intel/intel-device-plugins-for-kubernetes)
+
+  ```shell
+  kubectl apply -k 'https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/qat_plugin?ref=main'
+  ```
+
+  Verification of the plugin deployment and detection of QAT hardware can be confirmed by examining the resource allocations on the nodes:
+
+  ```shell
+  kubectl get node -o yaml| grep qat.intel.com
+  ```
+
+### For CryptoMB:
+
+It required the node with 3rd generation Intel Xeon Scalable processor server processors, or later.
+- For kubernetes Cluster, if not all nodes that support Intel® AVX-512 in Kubernetes cluster, you need to add some labels to divide these two kinds of nodes manually or using [NFD](https://github.com/kubernetes-sigs/node-feature-discovery).
+
+  ```shell
+  kubectl apply -k https://github.com/kubernetes-sigs/node-feature-discovery/deployment/overlays/default?ref=v0.15.1
+  ```
+
+- Checking the available nodes with required cpu instructions:
+  - Check the node labels if using [NFD](https://github.com/kubernetes-sigs/node-feature-discovery):
+
+    ```shell
+    kubectl get nodes -l feature.node.kubernetes.io/cpu-cpuid.AVX512F,feature.node.kubernetes.io/cpu-cpuid.AVX512DQ,feature.node.kubernetes.io/cpu-cpuid.AVX512BW,feature.node.kubernetes.io/cpu-cpuid.AVX512VBMI2,feature.node.kubernetes.io/cpu-cpuid.AVX512IFMA
+    ```
+
+  - Check CPUIDS manually on the node if without using NFD:
+
+    ```shell
+    cat /proc/cpuinfo |grep avx512f|grep avx512dq|grep avx512bw|grep avx512_vbmi2|grep avx512ifma
+    ```
+
+## Installation
+
+* Follow the steps from the [Quickstart Guide](../quickstart) to install Envoy Gateway.
+
+* Enables the EnvoyPatchPolicy feature.
```

Comment:
```suggestion
* Lets enable the EnvoyPatchPolicy feature, which will allow us to directly configure the Private Key Provider Envoy Filter, since Envoy Gateway does not directly expose this functionality.
```